src/hydro/hydro_driver.cpp

```diff
@@ -150,6 +226,47 @@ TaskCollection HydroDriver::MakeTaskCollection(BlockList_t &blocks, int stage) {
         hydro_pkg.get());
   }
 
+  // Calculate 1D profile of cooling rate
+  if (stage == 1) {
+    auto pkg = blocks[0]->packages.Get("Hydro");
+
+    AllReduce<PinnedArray1D<Real>> *pview_reduce =
+        pkg->MutableParam<AllReduce<PinnedArray1D<Real>>>("profile_reduce");
+
+    // initialize values to zero
+    for (int i = 0; i < pview_reduce->val.size(); i++) {
+      pview_reduce->val(i) = 0;
+    }
+
+    // create task region
+    int reg_dep_id;
+    TaskRegion &solver_region = tc.AddRegion(num_partitions);
+
+    for (int i = 0; i < num_partitions; i++) {
+      reg_dep_id = 0;
+      TaskList &tl = solver_region[i];
+
+      // compute rank-local reduction
+      auto &mu0 = pmesh->mesh_data.GetOrAdd("base", i);
+      TaskID local_sum =
+          (i == 0 ? tl.AddTask(none, CalculateCoolingRateProfile, mu0.get()) : none);
+
+      // NOTE: this is an *in-place* reduction!
+      TaskID start_view_reduce =
+          (i == 0 ? tl.AddTask(local_sum,
+                               &AllReduce<PinnedArray1D<Real>>::StartReduce,
+                               pview_reduce, MPI_SUM)
+                  : none);
+
+      // Test the reduction until it completes
+      TaskID finish_view_reduce =
+          tl.AddTask(start_view_reduce,
+                     &AllReduce<PinnedArray1D<Real>>::CheckReduce, pview_reduce);
+      solver_region.AddRegionalDependencies(reg_dep_id, i, finish_view_reduce);
+      reg_dep_id++;
```

Comment:
this may not even be required any more as the this is the end of the region (so the next region will first start once this is done), but I'll double check.

Reply:
yeah my understanding is also that this is not required.